libvirt/resource_libvirt_domain.go

```diff
@@ -344,16 +346,31 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 			if err != nil {
 				return fmt.Errorf("Error retrieving network name: %s", err)
 			}
+			dhcpMutex.Lock()
 			networkDef, err := newDefNetworkfromLibvirt(&network)
+			dhcpMutex.Unlock()
 			if !networkDef.HasDHCP() {
 				continue
 			}
 
 			hostname := domainDef.Name
+			hostnameSet := false
 			if hostnameI, ok := d.GetOk(prefix + ".hostname"); ok {
 				hostname = hostnameI.(string)
+				hostnameSet = true
```

Comment:
I would prefer that, but I remember that not working for some reason. When doing what you suggest, it seems golang doesn't want you to use `hostnameI` or `hostnameSet` outside of the `if` block.